modules/custom/d_p/d_p.module

```diff
@@ -244,6 +249,71 @@ function d_p_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   }
 }
 
+/**
+ * Move "heading_type" field from "Paragraph settings" under the main title.
+ *
+ * @param $form
+ *   The form that contains the d_settings field.
+ */
+function d_p_move_heading_type(&$form) {
+  if (isset($form["field_d_settings"]["widget"][0]["value"]) && isset($form["field_d_main_title"])) {
+    $widget = &$form["field_d_settings"]["widget"][0]["value"];
+    $widget['heading_type']['#weight'] = $form["field_d_main_title"]["#weight"] + 0.5;
+    $widget['heading_type']['#element_validate'] = ['d_p_heading_type_validate'];
+
+    // If field_d_main_title is in fieldgroup, locate it and move the heading_type under it.
+    if (isset($form["#group_children"]["field_d_main_title"])) {
+      $group = $form["#group_children"]["field_d_main_title"];
+      $form["#group_children"]['heading_type'] = $group;
+      $form[$group]['heading_type'] = $widget['heading_type'];
+    }
+    else {
+      $form['heading_type'] = $widget["heading_type"];
+    }
+    unset($widget["heading_type"]);
+  }
+}
+
+/**
+ * Custom validator for Heading Type.
+ *
+ * @param $element
+ * @param FormStateInterface $form_state
+ */
+function d_p_heading_type_validate($element, FormStateInterface $form_state) {
+  $parents = [];
+  foreach ($element["#parents"] as $index) {
+    if (!in_array($index, ['widget', 'heading_type', 'group_settings'], TRUE)) {
+      $parents[] = $index;
+    }
+  }
+  $parents[] = 'field_d_settings';
+  $current_settings = $form_state->getValue($parents);
+
+  // Inject heading_type settings into d_settings storage.
+  if (is_string($current_settings[0]['value'])) {
+    $decoded = json_decode($current_settings[0]['value'], TRUE) ;
+    $decoded['heading_type'] = $element["#value"];
+    $current_settings[0]['value'] = json_encode($decoded);
+  }
+  else {
+    $current_settings[0]['value']['heading_type'] = $element["#value"];
+  }
+  $form_state->setValue($parents, $current_settings);
+}
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function d_p_module_implements_alter(&$implementations, $hook) {
+  // Form alter hook runs as last.
+  if ($hook == 'form_alter') {
+    $group = $implementations['d_p'];
+    unset($implementations['d_p']);
+    $implementations['d_p'] = $group;
```

Comment:
hum.. mam tu zagwostkę.. albo to nic szczególnego nie zrobi, albo robi coś bardzo szczególnego :D

Reply:
Chodzi o to, że moduł field group ma swojego altera, który uruchamia się alfabetycznie, czyli po naszym z `d_p`. I jeśli nie przesunę naszego hooka na koniec to nie będę mieć jeszcze dostępnych grup :)